src/openms/source/CHEMISTRY/ProteaseDigestion.cpp

```diff
@@ -94,6 +95,20 @@ namespace OpenMS
   }
 
   Size ProteaseDigestion::digest(const AASequence& protein, vector<AASequence>& output, Size min_length, Size max_length) const
+  {
+    // initialization
+    output.clear();
+    std::vector<std::pair<size_t,size_t>> idcs; // small overhead filling intermediate vector first and iterating again
+    Size wrong_size = digest(protein, idcs, min_length, max_length);
+    std::transform(idcs.begin(), idcs.end(), output.begin(), 
+      [&protein](std::pair<size_t, size_t>& start_end){
+        protein.getSubsequence(start_end.first, UInt(start_end.second - start_end.first));
```

Comment:
isn't there a return needed?
and a std::back_inserter(output)?

Reply:
yes done.